### PR TITLE
add option --stop-on-failure

### PR DESCRIPTION
**Pull request by Franqsanz**
## 📚 Description

The `--stop-on-failure` option was supposed to be added, maybe it is not implemented correctly.
Problem described in #144 

## 🔖 Changes

Files changed:
[bashunit](https://github.com/TypedDevs/bashunit/blob/main/bashunit)
[runner.sh](https://github.com/TypedDevs/bashunit/blob/main/src/runner.sh)
[CHANGELOG.md](https://github.com/TypedDevs/bashunit/blob/main/CHANGELOG.md)

## ✅ To-do list

- [ ] Make sure that all the pipeline passes
- [ ] Make sure you have updated the documentation to reflect the changes or new features
- [ ] Make sure to update the `CHANGELOG.md` to reflect the new feature or fix

### CODE REVIEW

**Review by Tito-Kati — CHANGES_REQUESTED** [review]
Hello @Franqsanz , thank you very much for your interest in contributing to our project.

We have recently made some changes that cause conflicts with this PR, could you please resolve them?

Additionally, we need all features to have tests (we are a testing library, after all!), otherwise we cannot accept a PR.

Thank you very much again for helping us!

**Comment by Tito-Kati** [comment]
In the latest version, we have added a section in the documentation for the command line, you would need to add this feature to the docs as well.

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 - Add --version option
 - Add -v|--verbose option
 - Add ASCII art logo
+- Add option `--stop-on-failure`
```
Comment (Chemaclass) [inline CHANGELOG.md:18]: Can you move this line to the current `Unreleased` section (the one above) 👆

File: src/runner.sh [hunk 2]
```diff
@@ -123,6 +123,12 @@ function runner::run_test() {
   if [[ $test_result_code -ne 0 ]]; then
     state::add_tests_failed
     console_results::print_error_test "$function_name" "$test_result_code"
+
+    if [ "$_STOP_ON_FAILURE" = true ]; then
+      console_results::print_failed_test "$function_name" "$test_result_code"
+      exit 1
```
Comment (Chemaclass) [inline src/runner.sh:129]: Would it make sense to disable the exit trap before the exit 1, @Tito-Kati?
```suggestion
      trap '' EXIT && exit 1
```

File: CHANGELOG.md (hunk 1 above)
Comment (Franqsanz) [reply CHANGELOG.md:18]: Yes, for sure

File: src/runner.sh (hunk 2 above)
Comment (Chemaclass) [reply src/runner.sh:129]: Ok, we can do this change later if we see the need. We don't need to do it in this PR.

**Comment by Chemaclass** [comment]
I would like to have this before the weekend for the next release. Could you work on it, @Franqsanz? Otherwise, we will take over and work on it, it's fine :)

**Comment by Franqsanz** [comment]
Hi @Chemaclass, unfortunately I won't be able to work on this before the weekend, I'm sorry I couldn't finish this task, I've been busy with my work. 
 
Thanks for the patience and opportunity!  

Greetings!

**Pull request closed without merging**